ipatests/test_integration/test_adtrust_install.py

```diff
@@ -303,3 +304,570 @@ def test_ipa_user_s4u2self_pac(self):
+        """
+        This testcase checks that cldap responder doesnot hang
+        for domain discovery.
+        https://pagure.io/freeipa/issue/3639
+        """
+        version = tasks.get_openldap_client_version(self.master)
+        if version >= '2.6':
+            pytest.skip('bz2167328')
+        base_dn = str(self.master.domain.basedn)
+        srch_filter = "(&(DnsDomain='{}')(NtVer=\06\00\00\00)" \
+                      "(AAC=\00\00\00\00))".format(base_dn)
+        self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "-U"], raiseonerr=False
+        )
+        result = self.master.run_command(
+            ["ldapsearch", "-LL", "-H",
+             "cldap://{}".format(self.master.hostname),
+             "-b", "", "-s", "base", base_dn, srch_filter]
+        )
+        assert result.returncode == 0
+        assert 'dn: \n netlogon::*' in result.stdout_text
+
+    def test_user_connects_smb_share_if_locked_specific_group(self):
+        """
+        This testcase checks that with ipa-adtrust-installed and
+        the sidgen task registered and run, users cannot connect
+        a share in the local samba server if it is locked to specific
+        groups via "valid users", even if the user is a member of one
+        of those groups.
+        https://pagure.io/freeipa/issue/4234
+        """
+        msg = "tree connect failed: NT_STATUS_ACCESS_DENIED"
+        self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "-U"], raiseonerr=False
+        )
+        self.master.run_command(["mkdir", "/freeipa4234"])
+        self.master.run_command(
+            ["chcon", "-t", "samba_share_t",
+             "/freeipa4234"])
+        self.master.run_command(
+            ["setfacl", "-m", "g:admins:rwx",
+             "/freeipa4234"])
+        self.master.run_command(
+            ["net", "conf", "setparm", "share",
+             "comment", "Test Share"])
+        self.master.run_command(
+            ["net", "conf", "setparm", "share",
+             "read only", "no"])
+        self.master.run_command(
+            ["net", "conf", "setparm", "share",
+             "path", "/freeipa4234"])
+        self.master.run_command(["touch", "before"])
+        self.master.run_command(["touch", "after"])
+        self.master.run_command(
+            ["echo", "put", "before", "|",
+             "smbclient", "-k",
+             "//'{}'.format(self.master.hostname)/share"])
+        self.master.run_command(
+            ["net", "conf", "setparm", "share",
+             "valid users", "@admins"])
+        result = self.master.run_command(
+            ["echo", "put", "after", "|",
+             "smbclient", "-k",
+             "//'{}'.format(self.master.hostname)/share"])
+        assert msg not in result.stdout_text
+
+    def test_adtrust_install_without_ipa_installed(self):
```

Comment:
To speed up the test execution, you can move this test and test_samba_credential_cache_is_removed_post_uninstall just before test_adtrust_install_without_integrated_dns